gnocchi/metric/v1/resources/urls.go

```diff
@@ -11,6 +11,17 @@ func rootURL(c *gophercloud.ServiceClient, resourceType string) string {
 	return c.ServiceURL(resourcePath, resourceType)
 }
 
+func resourceURL(c *gophercloud.ServiceClient, resourceType, resourceID string) string {
+	if resourceType == "" {
+		resourceType = "generic"
+	}
```

Comment:
I think we should require the user to specify a resource type instead of providing a default of "generic". Having `""` as a parameter always looks strange.

Keep in mind that this package/library shouldn't be end-user facing, so the application which consumes this can always provide a default of "generic". Does that make sense? Thoughts?